pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -24,9 +24,11 @@ def __init__(self, pokemon, bot):
         self.response_key = ''
         self.response_status_key = ''
 
-    def work(self):
+    def work(self, response_dict=None):
```

Comment:
so i can call `create_encounter_api_call()`do something in between and then call work passing the returned response_dict.

#1887 does the same